File: app/src/main/java/me/ykrank/s1next/data/api/app/model/AppDataWrapper.java
package me.ykrank.s1next.data.api.app.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Created by ykrank on 2017/2/3.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AppDataWrapper<T> {
    @JsonProperty("data")
    private T data;
    private int code;
    private String message;
    private boolean success;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }
}

File: app/src/main/java/me/ykrank/s1next/data/api/app/model/AppDataWrapper.kt
package me.ykrank.s1next.data.api.app.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Created by ykrank on 2017/2/3.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
open class AppDataWrapper<T> : AppResult() {
    @JsonProperty("data")
    var data: T? = null
}
